usr/src/tools/svc/libscf/Makefile

```diff
@@ -59,7 +59,9 @@ LIBUUTIL =	$(SRC)/lib/libuutil
 CPPFLAGS +=	-DNATIVE_BUILD $(DTEXTDOM) \
 		-I$(SRC)/lib/libscf/inc -I$(COMDIR) -I$(LIBUUTIL)/common
 LDLIBS +=	-R '$$ORIGIN/../../lib/$(NATIVE_MACH)' -L$(ROOTONBLDLIBMACH) \
-		-luutil -lc -lgen -lnvpair -lsmbios
+		-luutil -lc -lgen -lnvpair
+$(NOT_AARCH64_BLD)LDLIBS += -lsmbios
+$(AARCH64_BLD)$(CROSS_BLD)LDLIBS += -lsmbios
```

Comment:
Why is this only needed in a cross build?

Reply:
How do you feel about

```diff
diff --git a/usr/src/tools/svc/libscf/Makefile b/usr/src/tools/svc/libscf/Makefile
index 8e1f3f84d3..a4605b271a 100644
--- a/usr/src/tools/svc/libscf/Makefile
+++ b/usr/src/tools/svc/libscf/Makefile
@@ -60,8 +60,10 @@ CPPFLAGS +=  -DNATIVE_BUILD $(DTEXTDOM) \
                -I$(SRC)/lib/libscf/inc -I$(COMDIR) -I$(LIBUUTIL)/common
 LDLIBS +=      -R '$$ORIGIN/../../lib/$(NATIVE_MACH)' -L$(ROOTONBLDLIBMACH) \
                -luutil -lc -lgen -lnvpair
-$(NOT_AARCH64_BLD)LDLIBS += -lsmbios
-$(AARCH64_BLD)$(CROSS_BLD)LDLIBS += -lsmbios
+
+i386_LDLIBS += -lsmbios
+LDLIBS += $($(NATIVE_MACH)_LDLIBS)
+
 ADJUNCT_LIBS +=        libsmbios.so libnvpair.so libgen.so libc.so

 .KEEP_STATE:
 ```

instead?

Reply:
This still works on a native build on `aarch64` and seems much cleaner.